lib.php

```diff
@@ -87,3 +87,47 @@ function auth_oidc_connectioncapability($userid, $mode = 'connect', $require = f
     }
     return $result;
 }
+
+function auth_oidc_before_footer() {
+    global $SESSION;
+    $oauth_origin = get_config('auth_oidc', 'oauthorigin');
+    $session_check_endpoint = get_config('auth_oidc', 'sessioncheckendpoint');
+    $logouturl = get_config('auth_oidc', 'logouturi');
+    $client_id = get_config('auth_oidc', 'clientid');
+    $session_state = isset($SESSION->session_state) ? $SESSION->session_state : '';
+?>
+    <iframe id="check-session-iframe" src="<?= $session_check_endpoint ?>" style="display: none"></iframe>
+    <script>
+        function checkSession(iframe, message, idpOrigin) {
+            iframe.contentWindow.postMessage(message, idpOrigin);
+        }
+
+        function handleSessionResponse(event, timerID, idpOrigin) {
+            if (event.origin !== idpOrigin) {
+                return;
+            }
+            if (event.data === "changed") {
+                if (timerID) {
+                    clearInterval(timerID);
+                }
+                window.location.replace("<?= $logouturl ?>");
+            }
+        }
+
+        function trackUserLoginStatus(iframe, client_id, session_state, idpOrigin) {
+            let timerID;
+            window.addEventListener("message", (event) => handleSessionResponse(event, timerID, idpOrigin), false);
+            var message = client_id + " " + session_state;
+            iframe.onload = () => {
+                checkSession(iframe, message, idpOrigin);
+                timerID = setInterval(() => checkSession(iframe, message, idpOrigin), 10 * 1000);
+            }
+        }
+
+        var session_state = "<?= $session_state ?>";
+        var client_id = "<?= $client_id ?>";
+        var iframe = document.getElementById("check-session-iframe");
+        trackUserLoginStatus(iframe, client_id, session_state, "<?= $oauth_origin ?>");
```

Comment:
Please make sure that the document is ready (iframe is rendered to DOM) before querying the iframe.